api/proto/config/storage.proto

```diff
@@ -193,3 +193,38 @@ message Volume {
   string displayName = 7;  // Optional friendly name echo'ed in info message
   bool clear_text = 8;  // Flag to indicate the volume encryption needed or not
 }
+
+// DiskConfigType is the desired configuration of disks
+enum DiskConfigType {
+  DISK_CONFIG_TYPE_UNSPECIFIED = 0; // no configured, do nothing
+  DISK_CONFIG_TYPE_EVEOS = 1; // the disk EVE is installed on
+  DISK_CONFIG_TYPE_PERSIST = 2; // the disk is separate persist partition, not zfs
+  DISK_CONFIG_TYPE_ZFS_ONLINE = 3; // included in zfs and online
+  DISK_CONFIG_TYPE_ZFS_OFFLINE = 4; // included in zfs and offline
+  DISK_CONFIG_TYPE_APPDIRECT = 5; // for direct assignment
+  DISK_CONFIG_TYPE_UNUSED = 6; // removed from zfs/app-direct
+}
+
+//DiskDescription describes disk
+message DiskDescription {
+  string disk_name = 1;
```

Comment:
Is this a bus-related name like /dev/sdc or a logical name line disk3?
I think we should add a logical_name field and add comments showing examples of the disk_name and logical_name.

Reply:
I put comments inside and add logical_name.